i3_agenda/API.py

```diff
@@ -67,65 +65,89 @@ def getEvents(
         page_token = calendar_list.get("nextPageToken")
         if not page_token:
             break
+    return calendar_ids
+
+
+def event_from_json(event):
+    end_time = get_event_time(event["end"].get("dateTime", event["end"].get("date")))
+    start_time = event["start"].get("dateTime", event["start"].get("date"))
+    unix_time = get_event_time(start_time)
+
+    location = None
+
+    if "location" in event:
+        location = event["location"]
+    elif "description" in event:
+        matches = re.findall(URL_REGEX, event["description"])
+        location = matches[0][0] if matches else None
+
+    return Event(
+            event["summary"],
+            is_allday(start_time),
+            unix_time,
+            end_time,
+            location,
+        )
+
+
+def get_event_result_today_only(service, calendar_id, max_results):
```

Comment:
a lot of code duplication from the `get_event_result` function, probably better to have a 3 function that receives a `timeMax` argument, this function will pass `midnight_rfc3339` to it, and the `get_event_result` will pass `None`.  
Also, after changing this function to match the behavior I described for the other function (return the `get(..)` value, you could just rename it to `get_today_events`

Reply:
I guess that it is done now.